src/mame/heathkit/tlb.cpp

```diff
@@ -1174,3 +1182,37 @@ ioport_constructor heath_gp19_tlb_device::device_input_ports() const
 {
 	return INPUT_PORTS_NAME(gp19);
 }
+
+
+heath_tlb_connector::heath_tlb_connector(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock) :
+	device_t(mconfig, HEATH_TLB_CONNECTOR, tag, owner, clock),
+	device_single_card_slot_interface(mconfig, *this),
+	m_write_sd(*this),
+	m_dtr_cb(*this),
+	m_rts_cb(*this),
+	m_reset(*this),
+	m_tlb(nullptr)
+{
+}
+
+heath_tlb_connector::~heath_tlb_connector()
+{
+}
+
+void heath_tlb_connector::device_start()
+{
+	m_tlb = get_device();
+}
+
+device_heath_tlb_card_interface *heath_tlb_connector::get_device()
+{
+	device_heath_tlb_card_interface *const connected = dynamic_cast<device_heath_tlb_card_interface *>(get_card_device());
+	if (connected)
+	{
+		return connected;
+	}
+	else
+	{
+		return nullptr;
+	}
+}
```

Comment:
This also shouldn’t be necessary when you’re deriving from `device_single_card_slot_interface`.